src/systems/systems/src/hydraulic/linear_actuator.rs

```diff
@@ -0,0 +1,1302 @@
+            signed_flow,
+            speed,
+        )
+    }
+
+    fn update_actions(&mut self, requested_mode: LinearActuatorMode, position_normalized: Ratio) {
+        match requested_mode {
+            LinearActuatorMode::ClosedValves => {
+                self.actions_from_current_to_closed_valves(position_normalized)
+            }
+            LinearActuatorMode::PositionControl => self.actions_from_current_to_position_control(),
+            LinearActuatorMode::ActiveDamping => self.actions_from_current_to_damping(),
+        }
+    }
+
+    fn actions_from_current_to_closed_valves(&mut self, position_normalized: Ratio) {
+        match self.current_mode {
+            LinearActuatorMode::ClosedValves => {}
+            LinearActuatorMode::PositionControl => {
+                self.go_to_close_control_valves(position_normalized);
+            }
+            LinearActuatorMode::ActiveDamping => {
+                self.go_to_close_control_valves(position_normalized);
+            }
+        }
+    }
+
+    fn actions_from_current_to_position_control(&mut self) {
+        match self.current_mode {
+            LinearActuatorMode::ClosedValves => {
+                self.go_to_position_control();
+            }
+            LinearActuatorMode::PositionControl => {}
+            LinearActuatorMode::ActiveDamping => {
+                self.go_to_position_control();
+            }
+        }
+    }
+
+    fn actions_from_current_to_damping(&mut self) {
+        match self.current_mode {
+            LinearActuatorMode::ClosedValves => {
+                self.go_to_damping();
+            }
+            LinearActuatorMode::PositionControl => {
+                self.go_to_damping();
+            }
+            LinearActuatorMode::ActiveDamping => {}
+        }
+    }
+
+    fn go_to_close_control_valves(&mut self, position_normalized: Ratio) {
+        self.closed_valves_reference_position = position_normalized;
+        self.current_mode = LinearActuatorMode::ClosedValves;
+    }
+
+    fn go_to_position_control(&mut self) {
+        self.current_mode = LinearActuatorMode::PositionControl;
+    }
+
+    fn go_to_damping(&mut self) {
+        self.current_mode = LinearActuatorMode::ActiveDamping;
+    }
+
+    fn force(
+        &mut self,
+        required_position: Ratio,
+        position_normalized: Ratio,
+        current_pressure: Pressure,
+        signed_flow: VolumeRate,
+        speed: Velocity,
+    ) -> Force {
+        match self.current_mode {
+            LinearActuatorMode::ClosedValves => {
+                self.force_closed_valves(position_normalized, speed)
+            }
+            LinearActuatorMode::ActiveDamping => self.force_damping(speed),
+            LinearActuatorMode::PositionControl => self.force_position_control(
+                required_position,
+                position_normalized,
+                signed_flow,
+                current_pressure,
+                speed,
+            ),
+        }
+    }
```

Comment:
**thought**: The name makes this function feel like a getter or something that calculates a value and returns it. This is true for `ClosedValves` and `ActiveDamping`. `PositionControl` however does adapt `self`. Should this function also be called `update_force`? Should an `update` function return a value, or should the updating and getting be separated?

Reply:
Sounded easy to use down into the LinearActuator but in fact just updating then getting is perfectly nice, and I removed the useless force from LinearActuator